dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,455 @@
+        Assert.assertNotNull(serviceIntegrationDetailedView.getSites());
+        Assert.assertFalse(serviceIntegrationDetailedView.getSites().isEmpty());
+        Assert.assertEquals(serviceIntegrationDetailedView.getSites().get(0).getId(),host.getIdentifier());
+
+        final Response deleteIntegrationsResponse = serviceIntegrationResource.deleteAllServiceIntegrationSecrets(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_OK, deleteIntegrationsResponse.getStatus());
+
+        final Response detailedIntegrationResponseAfterDelete = serviceIntegrationResource.getDetailedServiceIntegration(request, response, serviceKey, host.getIdentifier());
+        Assert.assertEquals(HttpStatus.SC_NOT_FOUND, detailedIntegrationResponseAfterDelete.getStatus());
+
+        //Now test the entry has been removed from the list of available configurations.
+        final Response hostIntegrationsResponseAfterDelete = serviceIntegrationResource.getServiceIntegrationByKey(request, response, serviceKey);
+        Assert.assertEquals(HttpStatus.SC_OK, hostIntegrationsResponseAfterDelete.getStatus());
+        final ResponseEntityView responseEntityViewAfterDelete = (ResponseEntityView) hostIntegrationsResponseAfterDelete.getEntity();
+        final ServiceIntegrationView serviceIntegrationHostViewAfterDelete = (ServiceIntegrationView) responseEntityViewAfterDelete.getEntity();
+      //  final ServiceIntegrationView serviceIntegrationViewAafterDelete = serviceIntegrationHostViewAfterDelete;
+        Assert.assertEquals(0, serviceIntegrationHostViewAfterDelete.getConfigurationsCount());
+        Assert.assertEquals("lola", serviceIntegrationHostViewAfterDelete.getName());
+        final List<SiteView> expectedEmptyHosts = serviceIntegrationHostViewAfterDelete.getSites();
+        Assert.assertNotNull(expectedEmptyHosts);
+        Assert.assertTrue(expectedEmptyHosts.isEmpty());
+    }
+
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration_Then_Delete_One_Single_Secret() throws IOException {
+
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true),
+                "param2", Param.newParam("val-2", false, Type.STRING, "label", "hint", true),
+                "param3", Param.newParam("val-3", false, Type.STRING, "label", "hint", true)
+        );
+        final Host host = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final String serviceKey = String.format("lol_%d",System.currentTimeMillis());
+        final String fileName = String.format("%s.yml",serviceKey);
+        final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)